### PR TITLE
Draft link refactor

### PR DESCRIPTION
**Pull request by MichaelChung123**
# Description
Making links open in a new tab for Submission Draft View.

## Types of changes
Re-structured the function to attach attributes, and moved functionality into lifecycle method in the Component.

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-205

File: app/frontend/src/utils/transformUtils.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { IdentityMode } from '@/utils/constants';
+import formioUtils from 'formiojs/utils';
```
Comment (jujaga) [inline app/frontend/src/utils/transformUtils.js:2]: Style: Normally we tend to import external libraries first before we import our internal codebase. Would be good to have this line be above Line 1.

File: app/frontend/src/utils/transformUtils.js [hunk 2]
```diff
@@ -42,3 +43,24 @@ export function parseIdps(identityProviders) {
   }
   return result;
 }
+
+/**
+ * @function attachAttributesToLinks
+ * Attaches attributes to <a> Link tags to open in a new tab
+ * @param {Component[]} formSchemaComponents An array of Components
```
Comment (jujaga) [inline app/frontend/src/utils/transformUtils.js:50]: IIRC the JSDoc doesn't support custom types like Component unless you define it somewhere else what that type is. I'd recommend changing this to just be `Object[]` and let the description handle describing what kind of object array it is.

File: app/frontend/src/utils/transformUtils.js [hunk 3]
```diff
@@ -42,3 +43,24 @@ export function parseIdps(identityProviders) {
...
+ * @returns {void} Does not return anything. Simply transforms the Components using FormIO Utils.
```
Comment (jujaga) [inline app/frontend/src/utils/transformUtils.js:51]: If there is no return value in a function, this line can be completely omitted.

**Review by jujaga — APPROVED** [review]
Looks good. Tentatively approving with some minor changes requested. 👍

**Comment by TimCsaky** [comment]
looks good. 
i forget why this function needs to be called beforeUpdate.. 

also I kind of wonder if we should have a 'do all formio fixes' function called something like 'processFormSchema()" in /utils/formShemaUtils.js. as there may be more of these fixes and it seems a bit unexpected to just call this minor change.

**Review by loneil — APPROVED** [review]

**Review by TimCsaky — APPROVED** [review]

**Comment by MichaelChung123** [comment]
Rabased

File: app/frontend/src/utils/transformUtils.js (hunk 1 above)
Comment (MichaelChung123) [reply app/frontend/src/utils/transformUtils.js:1]: oved this import in new commit

**Pull request merged**